csharp/src/Apache.Arrow/Ipc/ArrowReaderImplementation.cs

```diff
@@ -182,23 +185,46 @@ private void ReadDictionaryBatch(Flatbuf.DictionaryBatch dictionaryBatch, ByteBu
                 Flatbuf.FieldNode fieldNode = recordBatchEnumerator.CurrentNode;
 
                 ArrayData arrayData = field.DataType.IsFixedPrimitive()
-                    ? LoadPrimitiveField(ref recordBatchEnumerator, field, in fieldNode, messageBuffer)
-                    : LoadVariableField(ref recordBatchEnumerator, field, in fieldNode, messageBuffer);
+                    ? LoadPrimitiveField(ref recordBatchEnumerator, field, in fieldNode, messageBuffer, decompressor)
+                    : LoadVariableField(ref recordBatchEnumerator, field, in fieldNode, messageBuffer, decompressor);
 
                 arrays.Add(ArrowArrayFactory.BuildArray(arrayData));
             } while (recordBatchEnumerator.MoveNextNode());
 
             return arrays;
         }
 
+        private IBufferDecompressor GetDecompressor(BodyCompression? compression)
+        {
+            if (!compression.HasValue)
+            {
+                return new UncompressedBufferDecompressor();
```

Comment:
Name as NoOpDecompressor may be more clear than uncompressed decompressor. The current naming makes me ask how/why.